### PR TITLE
Adding prefect option

### PR DESCRIPTION
**Pull request by prasunanand**

### CODE REVIEW

File: qhub/template/{{ cookiecutter.repo_directory }}/.github/workflows/jupyterhub-pr.yaml [hunk 1]
```diff
@@ -68,3 +68,10 @@ jobs:
           tf_actions_subcommand: 'plan'
           tf_actions_working_dir: 'infrastructure'
           tf_actions_comment: true
+          {% if cookiecutter.prefect is defined %}
+          {% if cookiecutter.prefect %}
```
Comment (costrouc) [inline qhub/template/{{ cookiecutter.repo_directory }}/.github/workflows/jupyterhub-pr.yaml:72]: I think ` {% if cookiecutter.prefect %}` is redundant.

File: qhub/template/{{ cookiecutter.repo_directory }}/.github/workflows/jupyterhub.yaml [hunk 2]
```diff
@@ -68,10 +69,24 @@ jobs:
           tf_actions_subcommand: 'plan'
           tf_actions_working_dir: 'infrastructure'
           tf_actions_comment: true
+          {% if cookiecutter.prefect is defined %}
+          {% if cookiecutter.prefect %}
```
Comment (costrouc) [inline qhub/template/{{ cookiecutter.repo_directory }}/.github/workflows/jupyterhub.yaml:73]: I think {% if cookiecutter.prefect %} is redundant.

File: qhub/template/{{ cookiecutter.repo_directory }}/infrastructure/variables.tf [hunk 3]
```diff
@@ -85,3 +85,11 @@ variable "dask-worker-image" {
     tag  = "{{ cookiecutter.default_images.dask_worker.split(':')[1] }}"
   }
 }
+
+{% if cookiecutter.prefect is defined %}
+{% if cookiecutter.prefect %}
```
Comment (costrouc) [inline qhub/template/{{ cookiecutter.repo_directory }}/infrastructure/variables.tf:90]: I think {% if cookiecutter.prefect %} is redundant.

**Review by costrouc — CHANGES_REQUESTED** [review]

File: qhub/template/{{ cookiecutter.repo_directory }}/.github/workflows/jupyterhub-pr.yaml (hunk 1 above)
Comment (prasunanand) [reply qhub/template/{{ cookiecutter.repo_directory }}/.github/workflows/jupyterhub-pr.yaml:72]: What if we need to make it false ?

Comment (prasunanand) [reply qhub/template/{{ cookiecutter.repo_directory }}/.github/workflows/jupyterhub-pr.yaml]: Ok, will remove it

**Pull request merged**